core/src/main/java/nu/yona/server/subscriptions/service/UserPhotoService.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2017 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.subscriptions.service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import nu.yona.server.subscriptions.entities.UserPhoto;
+import nu.yona.server.subscriptions.entities.UserPhotoRepository;
+
+@Service
+public class UserPhotoService
+{
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private UserPhotoRepository userPhotoRepository;
+
+	@Transactional
+	public UserPhotoDto addUserPhoto(UUID userId, UserPhotoDto unsavedInstance)
```

Comment:
I don't see the added value of `unsaved` here either. It's the user photo that's to be added. It's a DTO and DTOs are never saved. We don't use this terminology for new goals, buddies or users either.

Reply:
Will rename